### PR TITLE
Comments, questions, and a bit of code cleanup

### PR DESCRIPTION
**Pull request by syoung-smallwisdom**
Sorry about the delay in getting a review back to you. I can't get it to build and might have more questions once I figure that out. Currently, this is just a lot of questions about naming conventions and whatnot.

### CODE REVIEW

File: shared/src/commonMain/kotlin/org/sagebionetworks/bridge/kmm/shared/apis/AssessmentsApi.kt [hunk 1]
```diff
@@ -7,6 +7,8 @@ package org.sagebionetworks.bridge.kmm.shared.apis
 import io.ktor.client.*
 import org.sagebionetworks.bridge.kmm.shared.models.AssessmentConfig
 
+// TODO: syoung 11/25/2020 Should the endpoint be a global constant?
```
Comment (nategbrown9) [inline shared/src/commonMain/kotlin/org/sagebionetworks/bridge/kmm/shared/apis/AssessmentsApi.kt:10]: Yes. Endpoint should be configurable here to allow for targeting test environments, but default should point to global constant.

File: shared/src/commonMain/kotlin/org/sagebionetworks/bridge/kmm/shared/apis/AuthenticationApi.kt [hunk 2]
```diff
@@ -27,6 +27,7 @@ internal class AuthenticationApi(basePath: kotlin.String = "https://webservices.
             level = LogLevel.ALL
             logger = object : Logger {
                 override fun log(message: String) {
+                    // TODO: syoung 11/25/2020 Given that this is an authentication service, is there a risk that the message will include PII?
```
Comment (nategbrown9) [inline shared/src/commonMain/kotlin/org/sagebionetworks/bridge/kmm/shared/apis/AuthenticationApi.kt:30]: Yes, need to remove PII before printing log message.

File: shared/src/commonMain/kotlin/org/sagebionetworks/bridge/kmm/shared/apis/DefaultHttpClient.kt [hunk 3]
```diff
@@ -27,6 +27,7 @@ object DefaultHttpClient {
             }
         }
         install(SessionTokenFeature) {
+            // TODO: syoung 11/25/2020 Should this be a global key? I notice that it is also used below.
```
Comment (nategbrown9) [inline shared/src/commonMain/kotlin/org/sagebionetworks/bridge/kmm/shared/apis/DefaultHttpClient.kt:30]: Yep, hadn't gotten around to moving it.

File: shared/src/commonMain/kotlin/org/sagebionetworks/bridge/kmm/shared/cache/AccountDao.kt [hunk 4]
```diff
@@ -20,6 +20,8 @@ class AccountDAO() {
 
     val hasSignedIn = reauthToken != null
 
+    // TODO: syoung 11/25/2020 Replace hardcoded appId? This looks like test test code.
+
```
Comment (nategbrown9) [inline shared/src/commonMain/kotlin/org/sagebionetworks/bridge/kmm/shared/cache/AccountDao.kt:24]: Yep, need to make it configurable.

File: shared/src/commonMain/kotlin/org/sagebionetworks/bridge/kmm/shared/cache/ResourceType.kt [hunk 5]
```diff
@@ -5,6 +5,8 @@ import kotlinx.serialization.decodeFromString
 import kotlinx.serialization.json.Json
 import kotlin.reflect.KClass
 
+// TODO: syoung 11/25/2020 I thought that the all caps with underscores was frowned upon? Shouldn't this be `AssessmentConfig`, etc.?
```
Comment (nategbrown9) [inline shared/src/commonMain/kotlin/org/sagebionetworks/bridge/kmm/shared/cache/ResourceType.kt:8]: For enums all caps is the standard: https://kotlinlang.org/docs/reference/enum-classes.html

File: shared/src/commonMain/kotlin/org/sagebionetworks/bridge/kmm/shared/cache/AccountDao.kt [hunk 6]
```diff
@@ -46,6 +48,7 @@ class AccountDAO() {
             }?: settings.remove(KEY_REAUTH_TOKEN)
         }
 
+    // TODO: syoung 11/25/2020 I thought that the all caps with underscores was frowned upon? Shouldn't this be `KeyEmail`, etc.? Also, the standard on Obj-c/Swift is "EmailKey" with the type of the object as the last bit. Is that different for Kotlin/Java?
```
Comment (nategbrown9) [inline shared/src/commonMain/kotlin/org/sagebionetworks/bridge/kmm/shared/cache/AccountDao.kt:51]: Constants should use uppercase underscore separated names: https://kotlinlang.org/docs/reference/coding-conventions.html#property-names

File: shared/src/commonMain/kotlin/org/sagebionetworks/bridge/kmm/shared/repo/AssessmentConfigRepo.kt [hunk 7]
```diff
@@ -39,6 +39,7 @@ class AssessmentConfigRepo(databaseDriverFactory: DatabaseDriverFactory) {
                 MainScope().launch {
 
                     val assessApi = assessmentsApi//loadAssessmentsApi(userSession.sessionToken)
+                    // TODO: syoung 11/25/2020 Is this method intentionally commented out rather than old code?
```
Comment (nategbrown9) [inline shared/src/commonMain/kotlin/org/sagebionetworks/bridge/kmm/shared/repo/AssessmentConfigRepo.kt]: Old code that I didn't clean up

File: shared/src/commonMain/kotlin/org/sagebionetworks/bridge/kmm/shared/repo/AssessmentConfigRepo.kt [hunk 8]
```diff
@@ -51,6 +52,7 @@ class AssessmentConfigRepo(databaseDriverFactory: DatabaseDriverFactory) {
                 }
 
                 //Need to make sure webcall is not already in flight
+                // TODO: syoung 11/25/2020 Is this comment a TODO?
```
Comment (nategbrown9) [inline shared/src/commonMain/kotlin/org/sagebionetworks/bridge/kmm/shared/repo/AssessmentConfigRepo.kt:57]: Yes, and I have code that handles this that just isn't checked in yet.

File: shared/src/commonMain/kotlin/org/sagebionetworks/bridge/kmm/shared/repo/AuthenticationRepository.kt [hunk 9]
```diff
@@ -21,6 +21,7 @@ class AuthenticationRepository {
         accountDAO.sessionToken = userSession.sessionToken
         accountDAO.reauthToken = userSession.reauthToken
 
+        // TODO: syoung 11/25/2020 Is this test code to get around consenting the user? Should it be here? And if so, please add a comment. Thanks!
```
Comment (nategbrown9) [inline shared/src/commonMain/kotlin/org/sagebionetworks/bridge/kmm/shared/repo/AuthenticationRepository.kt:24]: Yep, this is test code that needs a comment.

File: shared/src/commonMain/kotlin/org/sagebionetworks/bridge/kmm/shared/cache/ResourceType.kt (hunk 5 above)
Comment (syoung-smallwisdom) [reply shared/src/commonMain/kotlin/org/sagebionetworks/bridge/kmm/shared/cache/ResourceType.kt:8]: *sigh* That's inconsistent with sealed classes. Pity.

**Review by nategbrown9 — APPROVED** [review]

**Pull request merged**